### PR TITLE
DCT-135 NFT ALGO Guide

### PR DESCRIPTION
**Pull request by TheChronicMonster**
NFT Auction API Guide for Algo docs
Don't expect we'll want to merge this, but I think it's one of the better methods to revise before sending to Algorand

Includes complete guide for Algo and outline of our version of the tutorial.

### CODE REVIEW

File: docs/src/tut/nft/index.md [hunk 1]
```diff
@@ -0,0 +1,739 @@
...
+rather than determining where the application should access memory.
+
+Another reason is that most humans, even most engineers, 
+are not good at writing low-level code that is efficient, secure, and free of flaws. 
+Reach abstracts contract storage, protocol diagrams, state validation and network details so you can focus on how participants, users of the DApp, interact with one another and the contract.
+
+Reach's backend is very often one single Reach `rsh` file, which is syntactically the same as a JavaScript file, 
+but contains web3 functionality.
+The frontend in a Reach DApp can be created with JavaScript using Reach's [Standard JavaScript Library](https://docs.reach.sh/frontend/#ref-frontends).
+Alternatively, developers can use Python, Go, or C# in their frontend by utilizing Reach's [RPC Server](https://docs.reach.sh/rpc/).
```
Comment (jeapostrophe) [inline docs/src/tut/nft/index.md:52]:
```suggestion
Alternatively, developers can use Python, Go, or C# in their frontend by utilizing Reach's [RPC Server](https://docs.reach.sh/rpc/).
Or they can directly use the Algorand SDK and interact with their application using the standard Algorand ABI.
```

File: docs/src/tut/nft/index.md [hunk 2]
```diff
@@ -0,0 +1,739 @@
...
+
+Reach's backend, often named `index.rsh`, 
+instructs data to be published to the consensus network (a blockchain) and defines interfaces that can interact with the frontend.
```
Comment (jeapostrophe) [inline docs/src/tut/nft/index.md:55]: "instructs" is a weird word here. "orchestrates the data"?

File: docs/src/tut/nft/index.md [hunk 3]
```diff
@@ -0,0 +1,739 @@
...
+Reach abstracts contract storage, protocol diagrams, state validation and network details so you can focus on how participants, users of the DApp, interact with one another and the contract.
+
+Reach's backend is very often one single Reach `rsh` file, which is syntactically the same as a JavaScript file, 
+but contains web3 functionality.
+The frontend in a Reach DApp can be created with JavaScript using Reach's [Standard JavaScript Library](https://docs.reach.sh/frontend/#ref-frontends).
+Alternatively, developers can use Python, Go, or C# in their frontend by utilizing Reach's [RPC Server](https://docs.reach.sh/rpc/).
+
+Reach's backend, often named `index.rsh`, 
+instructs data to be published to the consensus network (a blockchain) and defines interfaces that can interact with the frontend. 
+This interface is commonly referred to as a Participant Interact Interface.
```
Comment (jeapostrophe) [inline docs/src/tut/nft/index.md:56]:
```suggestion
This interface is referred to as a Participant Interact Interface. 
```

File: docs/src/tut/nft/index.md [hunk 4]
```diff
@@ -0,0 +1,739 @@
...
+The frontend in a Reach DApp can be created with JavaScript using Reach's [Standard JavaScript Library](https://docs.reach.sh/frontend/#ref-frontends).
+Alternatively, developers can use Python, Go, or C# in their frontend by utilizing Reach's [RPC Server](https://docs.reach.sh/rpc/).
+
+Reach's backend, often named `index.rsh`, 
+instructs data to be published to the consensus network (a blockchain) and defines interfaces that can interact with the frontend. 
+This interface is commonly referred to as a Participant Interact Interface. 
+The frontend defines the interact objects and is often contained within a single file called `index.mjs`. 
+The frontend provides a graphical representation of the participants, e.g. Alice and Bob. 
+The user interface may be in the form of a command line, 
+or more preferably, a web or mobile app.
```
Comment (jeapostrophe) [inline docs/src/tut/nft/index.md:60]: It doesn't always provide a graphical representation. I'd say it provides an interface to the participants

File: docs/src/tut/nft/index.md [hunk 5]
```diff
@@ -0,0 +1,739 @@
...
+
+In Reach, developers think about their participants and how they will interact with one another and the contract. 
+Developers codify the business logic of their application and the Reach compiler assembles the bytecode and smart contract. 
+Reach abstracts the heavy lifting of smart contract development, 
+allowing developers to focus their energy on participant interaction. 
+
+Additionally, Reach's built in theorem verification engine ensures that the contract has the ability to exit,
```
Comment (jeapostrophe) [inline docs/src/tut/nft/index.md:67]:
```suggestion
Additionally, Reach's built-in formal verification engine ensures that the contract has the ability to exit, 
```

File: docs/src/tut/nft/index.md [hunk 6]
```diff
@@ -0,0 +1,739 @@
...
+
+In Reach, developers think about their participants and how they will interact with one another and the contract. 
+Developers codify the business logic of their application and the Reach compiler assembles the bytecode and smart contract. 
+Reach abstracts the heavy lifting of smart contract development, 
+allowing developers to focus their energy on participant interaction. 
+
+Additionally, Reach's built in theorem verification engine ensures that the contract has the ability to exit, 
+protects against common blockchain attacks, 
+guarantees the contract exits with a zero balance, 
+thus preventing locked away and inaccessible tokens,
```
Comment (jeapostrophe) [inline docs/src/tut/nft/index.md:70]:
```suggestion
guarantees the contract exits with a zero balance (thus preventing locked away and inaccessible tokens), 
```

File: docs/src/tut/nft/index.md [hunk 7]
```diff
@@ -0,0 +1,739 @@
...
+Reach abstracts the heavy lifting of smart contract development, 
+allowing developers to focus their energy on participant interaction. 
+
+Additionally, Reach's built in theorem verification engine ensures that the contract has the ability to exit, 
+protects against common blockchain attacks, 
+guarantees the contract exits with a zero balance, 
+thus preventing locked away and inaccessible tokens, 
+verifies the contract won't overspend, 
+confirms that formulas are true whether participants and frontends are honest or dishonest, 
+and facilitates token linearity.
```
Comment (jeapostrophe) [inline docs/src/tut/nft/index.md:73]:
```suggestion
and guarantees token linearity. 
```

File: docs/src/tut/nft/index.md [hunk 8]
```diff
@@ -0,0 +1,739 @@
...
+protects against common blockchain attacks, 
+guarantees the contract exits with a zero balance, 
+thus preventing locked away and inaccessible tokens, 
+verifies the contract won't overspend, 
+confirms that formulas are true whether participants and frontends are honest or dishonest, 
+and facilitates token linearity. 
+
+As a result, auditing Reach DApps is easier and less expensive when compared to low-level alternative programming languages. 
+Reach's compiler is more efficient than most human-built low-level code. 
+Altogether, Reach is the smartest, fastest, and safest way to build web3 decentralized applications.
```
Comment (jeapostrophe) [inline docs/src/tut/nft/index.md:77]: I bet they'd not want this sentence in it :)

File: docs/src/tut/nft/index.md [hunk 9]
```diff
@@ -0,0 +1,739 @@
...
+
+Reach backends are comprised of four primary modes: 
+* App Initialization
+* Local Step
+* Step
+* Consensus Step
+
+App Initialization refers to the section beginning at the creation of the Participant Interact Interface and ending at the `init()` statement. 
+
+A local step occurs on a participants local machine.
```
Comment (jeapostrophe) [inline docs/src/tut/nft/index.md:91]:
```suggestion
A local step occurs on a participant's local machine. 
```

File: docs/src/tut/nft/index.md [hunk 10]
```diff
@@ -0,0 +1,739 @@
...
+
+# Install Reach
+
+Reach is designed to work on POSIX systems with [make](https://en.wikipedia.org/wiki/Make_(software)), [Docker](https://www.docker.com/get-started), and [Docker Compose](https://docs.docker.com/compose/install/) installed. The best way to install Docker on Mac and Windows is with [Docker Desktop](https://www.docker.com/products/docker-desktop). 
+
+[Windows users](https://docs.reach.sh/guide/windows/#guide-windows) need to [install WSL2](https://docs.microsoft.com/en-us/windows/wsl/install-win10), a Windows subsystem for Linux, and run Reach through an Ubuntu command-line as administrator. 
+
+To confirm dependencies are installed, try to successfully run the following three terminal commands.
+
+
```
Comment (jeapostrophe) [inline docs/src/tut/nft/index.md:107]:
```suggestion

```

File: docs/src/tut/nft/index.md [hunk 11]
```diff
@@ -0,0 +1,739 @@
...
+```
+  $ make --version
+  $ docker --version
+  $ docker-compose --version
+```
+
+
```
Comment (jeapostrophe) [inline docs/src/tut/nft/index.md:114]:
```suggestion

```

File: docs/src/tut/nft/index.md [hunk 12]
```diff
@@ -0,0 +1,739 @@
...
+If you’re using Windows, consult [the guide to using Reach on Windows](https://docs.reach.sh/guide-windows.html).
+
+Next, make a directory for Reach and a subdirectory for the Auction DApp:
+
+
```
Comment (jeapostrophe) [inline docs/src/tut/nft/index.md:119]: Do that to the others unless this is their style

File: docs/src/tut/nft/index.md [hunk 13]
```diff
@@ -0,0 +1,739 @@
...
+```bash
+  $ ./reach update
+```
+
+
+## Setup environments
+
+Set an environment variable to use the Algorand Blockchain.
```
Comment (jeapostrophe) [inline docs/src/tut/nft/index.md:161]:
```suggestion
Set an environment variable to use the Algorand connector. 
```

File: docs/src/tut/nft/index.md [hunk 14]
```diff
@@ -0,0 +1,739 @@
...
+
+
+``` bash
+$ export REACH_CONNECTOR_MODE=ALGO
+```
+
+
+# Application Overview
+
+This Decentralized Auction involves a Creator who establishes the sale and Buyers who bid on an NFT in Algos.
```
Comment (jeapostrophe) [inline docs/src/tut/nft/index.md:171]:
```suggestion
This Decentralized Auction involves a Creator who establishes the sale and Buyers who bid on an NFT in ALGOs. 
```

Comment (jeapostrophe) [reply docs/src/tut/nft/index.md:171]: Same change everywhere else

File: docs/src/tut/nft/index.md [hunk 15]
```diff
@@ -0,0 +1,739 @@
...
++ Line 12: The final function, `showOutcome` which also takes an `Address` and `UInt` as inputs and has no output.
++ Line 14: Defines an `API` named `"Bidder"`
++ Line 15: Names `Bidder`'s only function, `bid`, which takes an unsigned integer as an input and outputs a `Tuple` containing an `Address` and `UInt`.
++ Line 17: Initializes the application and begins the first step of the Reach DApp.
+
+Now, let's set up the initial data definitions in the frontend.
+
+
+```
+load: /examples/nft-auction-api/index.mjs
```
Comment (jeapostrophe) [inline docs/src/tut/nft/index.md:242]: My guess is that when you send it over to Algorand, you'll need to include the raw code

**Review by jeapostrophe — CHANGES_REQUESTED** [review]

File: docs/src/tut/nft/index.md (hunk 8 above)
Comment (TheChronicMonster) [reply docs/src/tut/nft/index.md:77]: You should see what is in there, currently...

"With Reach, the automatic verification of a dApp, alone, is one great reason to consider using Reach vs PyTeal."

I toned the language down!

File: docs/src/tut/nft/index.md (hunk 15 above)
Comment (TheChronicMonster) [reply docs/src/tut/nft/index.md:228]: Yes, I'll do that.

**Comment by jeapostrophe** [comment]
I believe this was accepted into Algorand's dev portal and can be closed. Please reopen/etc if that's wrong.

**Pull request closed without merging**